src/Event/Decorator/ConditionDecorator.php

```diff
@@ -0,0 +1,73 @@
+<?php
+/**
+ * CakePHP : Rapid Development Framework (http://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ * @link          http://cakephp.org CakePHP Project
+ * @since         3.3.0
+ * @license       http://www.opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Cake\Event\Decorator;
+
+use Cake\Event\Event;
+use RuntimeException;
+
+/**
+ * Event Condition Decorator
+ */
+class ConditionDecorator extends AbstractDecorator
+{
+
+    /**
+     * @inheritdoc
+     * @return mixed
+     */
+    public function __invoke()
+    {
+        $args = func_get_args();
+        if (!$this->canTrigger($args[0])) {
+            return false;
```

Comment:
Won't this stop the event and prevent other listeners from being triggered? I would expect the condition/subject check to just make the decorated callable into a no-op not stop the event.